README.md

```diff
@@ -139,26 +139,129 @@ Dedicated servers hosted on Steam are usually required to be running the *latest
+
+#### Workdir
+
+The default working directory for all the images is [`/server`](build/Dockerfile#L70) within which all of a game's dedicated server files reside.
+
+#### Command line
+
+The following are some examples of how the game servers can be run:
+
+```shell
+# Counter-Strike: Global Offensive
+docker run -it -p 27015:27015/udp sourceservers/csgo:latest 'srcds_linux -game csgo -port 27015 +game_type 0 +game_mode 1 +mapgroup mg_active +map de_dust2'
+
+# Counter-Strike 1.6
+docker run -it -p 27016:27016/udp goldsourceservers/cstrike:latest 'hlds_linux -game cstrike +port 27016 +maxplayers 10 +map de_dust2'
+```
+
+* `-t` for a pseudo-TTY is mandatory; servers may not run correctly without it
+* `-i` for STDIN for interactive use of the game console
+* `-d` for running the container in detached mode
+
+#### Attaching
+
+If the game process is running as `PID 1` and STDIN is enabled for the container, the game's console can be accessed via:
+
+```shell
+docker attach containername
+```
+
+#### Debugging
+
+To debug a container or its files:
+
+```shell
+# To enter into a container
+docker exec -it containername bash
+
+# To issue detached command(s)
+docker exec containername ps aux                                     # Single, simple command
+docker exec containername bash -c 'printenv && ls -al && ps aux'     # Multiple or advanced commands
+```
+
+## Important considerations
+
+Due to the variety of `SRCDS` and `HLDS` games that can be hosted and the various ways each of the games can and/or have to be hosted, the images built using this project are kept to be as generic as possible. The following are important considerations concerning the images provided by the project.
+
+### Entrypoint script
+
+The game images **do not** include an entrypoint script.
+
+While the conventional `entrypoint.sh` could have been included, having so also takes away flexibility for how the images can be used. Operators wishing to utilize their own entrypoint scripts would have to include removal of pre-existing ones as part of their build or init process, which likely adds unnecessary confusion. It is also unlikely that a generic entrypoint script would be adequate given the various ways server operators could implement the container's initialization process for the game server.
+
+This brings us to the next but a much related consideration.
+
+### Environment variables
+
+The game images **do not** include support for mapped environment variables.
+
+Docker images are often designed with environment variables that are typically mapped onto arguments which are generated within the included entrypoint script as part of the container's initialization process that are then used for invoking the core binary for which the image is built.
```

Comment:
Include the fact that, environment variables may also be read directly by the application as configuration, for instance, [docker registry](https://docs.docker.com/registry/configuration/#override-specific-configuration-options) that comply with the [12 factor methodology - Environment as config](https://12factor.net/config).

Reply:
Could you pull and do a write up on it instead? 

`git fetch https://github.com/joeltimothyoh/docker-sourceservers.git chore/update-readme-with-usage-and-considerations:chore/update-readme-with-usage-and-considerations`

Reply:
@joeltimothyoh  u mean to open another PR ?

Reply:
No, you can commit straight into my fork which updates this PR. @leojonathanoh

Reply:
@joeltimothyoh 
done